js/background.js

```diff
@@ -1,80 +1,76 @@
 
-var Domain = "https://channeli.in";
-var Host = "channeli.in";
+var NET_CHECK_DOMAIN = "http://info.cern.ch/";
+var DOMAIN = "https://channeli.in";
+var HOST = "channeli.in";
 
-var getDomainName = function (href) {
-  var l = document.createElement("a");
-  l.href = href;
-  return l.hostname;
-}
+/** 0 implies 'not logged in' and 1 implies 'logged in' */
+var userStatus = -1;
+/** 0 implies 'not connected' and 1 implies 'connected' */
+var networkStatus = 0;
 
-var checkSession = function() {
-  var url = Domain + "/check-session/";
-  $.post(url, function(res){
-    if(res.msg == "YES") {
-      chrome.browserAction.setIcon({path: "../images/icon_active.png"});	
-    }
-    else if(res.msg == "NO") {
-      chrome.browserAction.setIcon({path: "../images/icon_inactive.png"});
+/**
+ * Check session if the changed tab was one of Channel-i
+ * @param tabId - the ID of the tab that is changing or has changed
+ * @param changeInfo - whether the tab is changing or has changed
+ * @param tab - the tab that is changing or has changed
+ */
+function updateListener(tabId, changeInfo, tab) {
+    var url = tab.url;
+    if (url !== undefined && changeInfo.status === "complete") {
+        var host = getHostName(url);
+        if (host == HOST) {
+            checkSession();
+        }
     }
-  });
 }
 
-// This is to handle the status updation when the app is "loaded" or "refreshed". 
-checkSession(); 
+/**
+ * Get the host name from a given URL
+ * @param href - the URL taken from the address bar
+ * @returns {string} hostname - the host name extracted from the URL
+ */
+var getHostName = function (href) {
+    var l = document.createElement("a");
+    l.href = href;
+    return l.hostname;
+};
 
-/* Updates the status especially when the 'channeli.in' tabs are updated */
-chrome.tabs.onUpdated.addListener(function(tabId, changeInfo, tab) {
-  var tab_url = tab.url;
-  var domain = getDomainName(tab_url);
-  if (tab_url !== undefined && changeInfo.status == "complete") {
-    if(domain == Host) {
-      checkSession();
-      /*****************************************************************************************
-      //TODO: When more than one tab from 'channel.in' domain are active, then logging out
-      //      from channeli in one tab should automatically make other tabs from the same  
-      //      domain to refresh along with the extension.
-
-      chrome.tabs.query({}, function (tabs) {
-          var _tabs = [];
-          for (var i = 0; i < tabs.length; i++) {
-            if(getDomainName(tabs[i].url) == Host)
-            {
-              if(tabs[i].id == tabId) continue; 
-              else {
-                _tabs.push(tabs[i].id);
-              }
-            }
-          }
-          for(var j = 0; j < _tabs.length; j++)
-          {
-            
-            chrome.tabs.reload(_tabs[j]);
-          }
-      });
-      ******************************************************************************************/
```

Comment:
I managed to do this! If the tab logs out without AJAX, other pages are instantly refreshed to log them out as well and if the tabs logs out via AJAX other tabs are refreshed in the next minute as a part of the routine checkSession.